File: backend/credit-wizard-api/credit-wizard-api/Dtos/DegreeModulDto.cs
namespace credit_wizard_api.Dtos;

public class DegreeModulDto
{
    public Guid ModulId { get; set; }

    public Guid DegreeId { get; set;}

    public ModulDto ModulDto { get; set; }
    public DegreeDto DegreeDto { get; set; }

    public bool IsRequired { get; set; } = false;
}

File: frontend/credit-wizard/src/app/shared/dtos/degreeModulDto.ts
import { Guid } from "guid-typescript";
import { IDegreeDto } from './degreeDto';
import { IModulDto } from "./modulDto";

export interface IDegreeModulDto {
    modulId: Guid,
    degreeId: Guid,
    modulDto: IModulDto,
    degreeDto: IDegreeDto,
    isRequired: boolean
}
